lib/progressive_hash_diff.rb

```diff
@@ -0,0 +1,30 @@
+class ProgressiveHashDiff
```

Comment:
We did have actually have a EditionActionFactory for a few days

Fluid interface doesn't work unless you save the variables, which is what we worked at avoiding.

eg this crude pseudo code
```
# start of command
edition_before_command = turn_edition_into_comparable_hash(edition)
# change edition in lots of ways over many lines and methods
edition_after_command = turn_edition_into_comparable_hash(edition)
# create and save diff
```

whereas this class encapsulates the representation of the item as a hash and stores the variables for you